WordPress/src/main/java/org/xmlrpc/android/XMLRPCUtils.java

```diff
@@ -313,13 +313,25 @@ private static String discoverSelfHostedXmlrpcUrl(String siteUrl, String httpUse
         // add the url as provided by the user
         urlsToTry.add(siteUrl);
 
-        // add a sanitized version of the url
-        final String sanitizedURL = sanitizeSiteUrl(siteUrl);
-        if (!urlsToTry.contains(sanitizedURL)) {
-            urlsToTry.add(sanitizedURL);
+        // add a sanitized version of the http url (if the user didn't specify it)
+        final String sanitizedHttpsURL = sanitizeSiteUrl(siteUrl, true);
+        if (!urlsToTry.contains(sanitizedHttpsURL)) {
+            urlsToTry.add(sanitizedHttpsURL);
+        }
+
+        String appendedXmlrpcUrl = appendXMLRPCPath(sanitizedHttpsURL);
+        if (!urlsToTry.contains(appendedXmlrpcUrl)) {
+            appendedXmlrpcUrl += "?rsd";
```

Comment:
Right, because `verifyXmlrpcUrl` tries to append `xmlrpc.php` and does POST.
The function here is used to discovery the endpoint by doing all the steps, in the correct order (without shortcuts). @hypest could confirm this, since he refactored all of my code about the new login.